tests/check/test_check_project_metadata.py

```diff
@@ -0,0 +1,226 @@
+        languages from LANGUAGE_DATA_EXTRACTION_DIR
+        """
+        available_languages = get_available_languages()
+        desired_dict = {
+            "arabic": {},
+            "estonian": {},
+            "indonesian": {},
+            "norwegian": {"sub_languages": ["bokmål", "nynorsk"]},
+            "spanish": {},
+            "basque": {},
+            "finnish": {},
+            "italian": {},
+            "persian": {},
+            "swahili": {},
+            "bengali": {},
+            "french": {},
+            "japanese": {},
+            "pidgin": {"sub_languages": ["nigerian"]},
+            "swedish": {},
+            "chinese": {"sub_languages": ["mandarin"]},
+            "german": {},
+            "korean": {},
+            "polish": {},
+            "tajik": {},
+            "czech": {},
+            "greek": {},
+            "kurmanji": {},
+            "portuguese": {},
+            "tamil": {},
+            "dagbani": {},
+            "hausa": {},
+            "latin": {},
+            "punjabi": {"sub_languages": ["gurmukhi", "shahmukhi"]},
+            "ukrainian": {},
+            "danish": {},
+            "hebrew": {},
+            "latvian": {},
+            "russian": {},
+            "yoruba": {},
+            "english": {},
+            "hindustani": {"sub_languages": ["hindi", "urdu"]},
+            "malay": {},
+            "sami": {"sub_languages": ["northern"]},
+            "esperanto": {},
+            "igbo": {},
+            "malayalam": {},
+            "slovak": {},
+        }
+        for lang in available_languages.keys():
+            self.assertIn(lang, desired_dict.keys())
+            if len(available_languages[lang]) > 0:
+                for sub_lang in available_languages[lang]["sub_languages"]:
+                    self.assertIn(sub_lang, desired_dict[lang]["sub_languages"])
+        self.assertEqual(len(available_languages), len(desired_dict))
+
+    def test_get_missing_languages(self):
+        """
+        Tests that get_missing_languages returns a list of languages missing given
+        a target language dictionary to check for and a reference language
+        dictionary to check in
+        """
+        reference_languages = {
+            "estonian": {},
+            "indonesian": {},
+            "norwegian": {"sub_languages": ["bokmål", "nynorsk"]},
+        }
+
+        target_languages = {
+            "estonian": {"iso": "et", "qid": "Q9072"},
+            "indonesian": {"iso": "id", "qid": "Q9240"},
+            "norwegian": {
+                "sub_languages": {
+                    "bokmål": {"iso": "nb", "qid": "Q25167"},
+                    "nynorsk": {"iso": "nn", "qid": "Q25164"},
+                }
+            },
+        }
+
+        # no missing languages
+        self.assertEqual(
+            get_missing_languages(reference_languages, target_languages), []
+        )
+
+        # more reference languages than target languages; should have no effect
```

Comment:
Minor comment here @GraceBocek, that for comments we try to have those on their own line be fully punctuated sentences including capitalization. So if it's its own line, then capitalize and include a period, and if it's inline then no capitalization and no period :) I'll get these this time!